resources/todomvc/architecture-examples/react-redux/src/containers/main-container.jsx

```diff
@@ -0,0 +1,23 @@
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import Main from "../components/main";
+import * as TodoActions from "../actions";
+import { withRouter } from "react-router-dom";
+import { getCompletedTodos, getVisibleTodos } from "../selectors/filters";
+
+const mapStateToProps = (state, ownProps) => {
+    const { todos } = state;
+    const { location } = ownProps;
+
+    const visibleTodos = getVisibleTodos(todos, location.pathname);
+    const completedCount = getCompletedTodos(todos).length;
+    const activeCount = todos.length - completedCount;
+
+    return { todos, completedCount, activeCount, visibleTodos };
+};
+
+const mapDispatchToProps = (dispatch) => ({
+    actions: bindActionCreators(TodoActions, dispatch),
+});
```

Comment:
optional nit: react-redux knows how to bind automatically and wrap `dispatch` (like you did for the header container above). I believe this would work (haven't checked):
```suggestion
const mapDispatchToProps = actions;
```